dm/dm/worker/subtask.go

```diff
@@ -257,6 +261,38 @@ func (st *SubTask) run() {
 	go cu.Process(ctx, pr)
 }
 
+func (st *SubTask) StartValidator(expect pb.Stage) {
+	st.Lock()
+	defer st.Unlock()
+	if st.cfg.ValidatorCfg.Mode != config.ValidationFast && st.cfg.ValidatorCfg.Mode != config.ValidationFull {
+		return
```

Comment:
seems this line should not happen, becasue we always adjust the subtaskConfig

Reply:
Adjusting the `subtaskConfig` seems insufficient to guarantee the `mode` is either `fast` or `full` since there are three possible values for it: `none`, `fast`, and `full`, where `none` is the default value (when this field is missing) and indicates the validator should not start.

Reply:
> seems this line should not happen, becasue we always adjust the subtaskConfig

`st.cfg.ValidatorCfg.Mode` may equals to `none`(i.e. validation is disabled)